test/files/run/t6714.scala

```diff
@@ -0,0 +1,29 @@
+
+case class A(a: Int, index: Int) {
+  def apply(i: Int)(implicit ev: Int): A = new A(ev, i)
+  def update(i: Int, value: Int): A = if (i == index) A(i, value) else A(i, 0)
+  def +(j: Int) = a + j
+}
+
+case class B(a: Int, index: Int) {
+  def apply(i: Int)(j: Int)(implicit ev: Int): B = new B(j + ev, i)
+  def update(i: Int, value: Int): B = if (i == index) B(i, value) else B(i, 0)
+  def +(j: Int) = a + j
+}
+
+object Test extends App {
+  def checkA(x: A, y: Int, z: Int) = assert(x == A(y, z))
+  def checkB(x: B, y: Int, z: Int) = assert(x == B(y, z))
+  implicit def ev: Int = 8000
+  val a = A(5, 1)
+  checkA(a(1) = 10, 1, 10)
+  checkA(a(1) += 20, 1, 8020)
+  val b = B(5, 1)
+  checkB(b(1)(5) += 20, 1, 8025)
```

Comment:
It's not allowed by the spec. I'd be comfortable failing in the new edge cases, too, per Seth's reaction on the second ticket.

Regular update for `v(1) = rhs` is done by parser. For `v(1) += rhs`, with an implicit arg, so we see `v(1)(imp) += rhs`, do we allow only if the arg was implicitly supplied, in order to enforce a syntactic rewrite? This PR just takes the indices to update from the first arg list and takes the whole LHS for the apply expression. `v.update(1, v(1)(anything)(imp) + rhs)` if there are more arglists.

The current just drops args, which is wrong.